src/token/api/src/canister/inspect.rs

```diff
@@ -49,53 +48,55 @@ static TRANSACTION_METHODS: &[&str] = &[
     "transferIncludeFee",
 ];
 
+/// Reason why the method may be accepted.
+#[derive(Debug, Clone, Copy)]
+pub enum AcceptReason {
+    /// The call is a part of the IS20 API and can be performed.
+    Valid,
+    /// The method isn't a part of the IS20 API, and may require further validation.
+    NotIS20Method,
+}
+
 /// This function checks if the canister should accept ingress message or not. We allow query
 /// calls for anyone, but update calls have different checks to see, if it's reasonable to spend
 /// canister cycles on accepting this call. Check the comments in this method for details on
 /// the checks for different methods.
-#[cfg(not(feature = "no_api"))]
-#[inspect_message]
-fn inspect_message() {
-    let method = ic_cdk::api::call::method_name();
-
-    let state = CanisterState::get();
-    let state = state.borrow();
-    let caller = ic_cdk::api::caller();
-
-    match &method[..] {
+pub fn inspect_message(
+    state: &CanisterState,
+    method: &str,
+    caller: Principal,
+) -> Result<AcceptReason, &'static str> {
+    match method {
         // These are query methods, so no checks are needed.
-        "mint" if state.stats.is_test_token => ic_cdk::api::call::accept_message(),
-        m if PUBLIC_METHODS.contains(&m) => ic_cdk::api::call::accept_message(),
+        "mint" if state.stats.is_test_token => Ok(AcceptReason::Valid),
+        m if PUBLIC_METHODS.contains(&m) => Ok(AcceptReason::Valid),
```

Comment:
If we use exclude these methods from the API by using the features, we probably don't wont the logic for their inspection to be here also.

Reply:
would be nice in that case if the inspect message methods could be imported separately and compiled as one.

Reply:
Inspections are disabled for the excluded methods now.